File: tkpdflight/src/main/java/com/tokopedia/flight/FlightComponentInstance.java
package com.tokopedia.flight;

import android.app.Application;

import com.tokopedia.abstraction.base.app.BaseMainApplication;
import com.tokopedia.flight.common.di.component.FlightComponent;
import com.tokopedia.flight.common.di.component.DaggerFlightComponent;

/**
 * Created by nakama on 11/12/17.
 */

public class FlightComponentInstance {
    private static FlightComponent flightComponent;

    public static FlightComponent getFlightComponent(Application application) {
        if (flightComponent == null) {
            flightComponent = DaggerFlightComponent.builder().baseAppComponent(
                    ((BaseMainApplication)application).getBaseAppComponent()).build();
        }
        return flightComponent;
    }
}

File: features/travel/flight/src/main/java/com/tokopedia/flight/FlightComponentInstance.kt
package com.tokopedia.flight

import android.app.Application
import com.tokopedia.abstraction.base.app.BaseMainApplication
import com.tokopedia.flight.common.di.component.DaggerFlightComponent
import com.tokopedia.flight.common.di.component.FlightComponent

/**
 * Created by nakama on 11/12/17.
 */
object FlightComponentInstance {
    private var flightComponent: FlightComponent? = null

    @JvmStatic
    fun getFlightComponent(application: Application): FlightComponent? {
        if (flightComponent == null) {
            flightComponent = DaggerFlightComponent.builder().baseAppComponent(
                    (application as BaseMainApplication).baseAppComponent).build()
        }
        return flightComponent
    }
}